### DISCUSSION TITLE
where has the theme-chooser gone !? is it still in use !?

### DISCUSSION THREAD

**Discussion by fsj-digital** [community/community#33924] [Other Feature Feedback, Questions, & Ideas]
hi there dear community, 

i am currenty setting up a page - and i wonder where the theme-chooser has gone. 

according this instruction here - i want to go through the process: 

![image](https://user-images.githubusercontent.com/32641183/191938623-ea79e3fe-8bcc-4b8c-aa9f-3aa98d1be604.png)

is this doucumentation still correct and are the information still relevant!? 

![image](https://user-images.githubusercontent.com/32641183/191939026-07a54846-26ba-45c5-bf99-4c3382f2f320.png)

note: i also wanna **set up a jekyll minimal theme** like carolyn does it here https://carolstran.github.io/cv/

```
Under Source, it should say, “GitHub Pages is currently disabled. You must first add content to your repository before you can publish a GitHub Pages site.” Because your repository is empty, you have no branches and you’re unable to select a source. That’s ok! This will be taken care of momentarily.

The subsection you want is Theme Chooser. Here, you’ll select a design template for your site. This will then create a gh-pages branch and publish your site with your selected theme from that branch. 

All of the options are [Jekyll](https://jekyllrb.com/) themes. If you don’t know Jekyll, that’s ok. This guide will walk you through the necessary configuration. Honestly, I have no idea how Jekyll works beyond GitHub Pages templates.

Select a theme (design template) for your CV
CVs are supposed to give future employers an idea of who you are - so choose a theme that speaks to you! For this guide, we’ll continue with the “Minimal” theme as an example because that’s the one I use for [my CV](https://carolstran.github.io/cv/).

```

note: i also wanna **set up a jekyll minimal theme** like carolyn does it here https://carolstran.github.io/cv/

any idea

**Comment by radar-blip** [comment]
Somebody asked this yesterday also. 
I'm not sure why, but for some reason it isn't there anymore.

**Comment by yoannchaudet** [comment]
We did [deprecate the theme chooser](https://github.blog/changelog/2022-08-22-github-pages-deprecating-the-theme-picker/).

Themes are still supported of course and you can check the previous link to see how to set one up.

**Reply by sauravwel** [reply]
Thanks man for info.

**Reply by nipunikajain** [reply]
you shouldn't have deprecated it.

**Reply by playg8** [reply]
This is not right. I don't want to learn jekyll at all, no time. 
Very very bad idea for remove it out.

**Reply by Yarroudh** [reply]
I think this is a bad Idea. The themes were very useful.

**Reply by ncuxie** [reply]
Very very bad idea，Very inconvenient

**Reply by AMark-CS** [reply]
very very bad idea, I do hope it can be added again

**Reply by niax69** [reply]
A shit what you have done.

**Reply by EricXiongLi** [reply]
Very bad idea

**Reply by dtap001** [reply]
noo

**Reply by OS-Scratch** [reply]
oh no... but JEKYLL is soo hard...

**Reply by DanKonigsbach** [reply]
In that case, https://pages.github.com/ should be updated to reflect the new approach. Until that is done, it only leads the interested beginner down a path of frustration and failure.

(I don't volunteer to author the update because I don't know the correct, new approach. Someone who does would need to update it.)

**Comment by houmie** [comment]
I wished you hadn't done this.  What did you gain from deprecating this?  It was so easy and nice quickly creating a project page.
Now it is so much more difficult. Please try to sit down and follow the steps here for Mac: https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll

You will come across multiple issues with Jekyll, Ruby, Bundler and the steps are often inaccurate too.  

If you needed to create something quickly without local dependencies, the 'Choose Theme' on server side was so perfect for the job.  What a shame.

**Comment by tkolarik** [comment]
If the feature has been removed, you should probably remove it from the instructions here https://pages.github.com/

**Reply by 13x4** [reply]
Yeah it totally misleading and useless for it to appear as helping context for choosing themes over here. Don't have enough time to learn and use Jekyll or whatever on Earth this stuff is. :/

**Reply by eloyhz** [reply]
This is the new feature https://github.com/skills/github-pages

**Reply by kairin** [reply]
it is 2024... and i realised this is mistake because nothing on the  https://pages.github.com/ indicates that this option has been deprecated. 

come on guys...

at least indicate somewhere on this URL...

![image](https://github.com/user-attachments/assets/0bf64fd7-050e-4201-a489-a77e322830c3)

**Comment by zanzix** [comment]
Ok so - the theme chooser was quietly removed, the documentation for setting up github pages still tells you to use the non-existent theme chooser, no one has bothered to update this documentation despite it being months out of date, and the new documentation is an encyclopedia page that drowns you in superfluous information about something that used to be as easy as a single click of a button. 

Thank you, Microsoft, for making sure that I will always have job security as a front-end developer. 👍

**Reply by yoannchaudet** [reply]
https://github.blog/changelog/2022-08-22-github-pages-deprecating-the-theme-picker/

Our [marketing page](https://pages.github.com/) was not updated yet because there is more to come. The documentation was. If you find something that needs an update in the documentation, please share the link.

You are very much welcome.

**Reply by OscarAbraham** [reply]
@yoannchaudet I realize that updating marketing pages may not be the job of the documentation team at GitHub. I don't mean any disrespect to GitHub's documentation; it's generally very good and very thorough.

The thing is these "marketing pages" really, really look and behave like documentation. Most of what they contain is basic instructions for using a product. They are also often at the top of web searches. This results in an onboarding nightmare when their information is so obsolete. These "marketing pages" are misleading beginners, and those are the users that need clear information the most. In this case, the "marketing" for GitHub Pages clearly seems to suggest that the most practical way for creating a site from a repo is one that no longer exists.

I'm not a marketing expert, but, as a user, it seems like a bad idea to have marketing pages with misleading information for months, or even years, while "more is to come". It's frustrating, and it creates a bad image for the product and for GitHub. This not the only case where there's a problem like this; for example, the [page for GitHub for Unity](https://unity.github.com/) leads to an abandoned, unusable project that no one should use. Even if these pages are not considered "documentation", it's still wrong to leave incorrect information for beginners to find at the top of web searches.

**Reply by DavidBradbury** [reply]
Hi there - It's 9, almost 10 months later - I just wasted an hour trying to figure this out because you haven't updated https://pages.github.com/ - Nobody sees that as marketing, they see it as the first page you look at when going to set up pages for the first time. Please provide a redirect if you're not going to update this.

**Reply by johnbede2000** [reply]
totally agree, the 'marketing' pages are presented as instructions and any Googler comes across these instructions first. I was trying to follow them and wasted time. I had to guess that it's been deprecated in order to be able to google that and find confirmation for my guess

**Reply by karinabao** [reply]
It's now July 2024 and [https://pages.github.com/](https://pages.github.com/) is still not updated. Disappointed.

**Reply by thabjeel** [reply]
this is still not updated.

**Reply by Wiilly-i-am** [reply]
17/02/2025:23:06:40 and nothing has changed.

**Comment by sciustechnologia** [comment]
why take out that one thing that might have worked?
I am trying now a week 
1. getting Jekyll running on Mac (always ruby issues and the answers on the web are even more confusing)
2. finding the right ruby- Jekyll version (# jekyll 4.3.2, or version 3.9.2)combination in which folder (Cellar or lib) to get one theme running
3. authentication issues
4. run the server local (Liquid Exception: undefined method `tainted?') - impossible. This issue seems to a depending on compatible version as well.

I encountered similar issues on my Linux machine till the machine just went black screen.

**Reply by houmie** [reply]
This is how products die. Bad decisions by someone in the company, without thinking this through.  They haven't even sat down once to follow the steps locally from scratch and trying to see how difficult it is to get Jekyll up and running.

I can highly recommend switching to Hugo and integrate it with Netlify to Github.  Hugo has many beautiful themes and it's very easy to deploy everything within 10 min on GitHub.

**Reply by sciustechnologia** [reply]
this is what I learned so far:

1. the happy path documentation for installation is pure marketing and has nothing to do with user-friendly installation. 
    Common issue: No or only rudimentary information about requirements, for instance.
3. installing Jekyll on the local machine (Mac, Linux) is very difficult since the entire installation is sooooo compartmentalized, that versions are prone conflict. Mac and Ruby are not not matching, brew and ruby and not matching, Liquid and the entire system are not matching, and so on
4. themes are few but/and the documentation is often insufficient and/or not updated, ever. I tried all themes (https://pages.github.com/themes/). This needs to be done better, way better.
5.  If there is someone testing before release, what version worked for them (him, her, them)?

Github is hurting its own reputation by publishing software that is not working - after they took out the link that would do the installation of the theme automatically. Not only is this a missed opportunity to enhance but also counterproductive. Frustrating your customers doesn't make you more popular. 


I wonder 
1. why is github involved in this? 
2. are there users who got it running and displayed on GH-pages?
PLEASE, share your work. I need to believe that this is working. And I need to see it to believe it.

**Reply by playg8** [reply]
I am very sad. Watching at the document again and again, also can not get it. Jekyll is really a bad tool for me.

**Comment by james-s-tayler** [comment]
I can only guess that it's secretly because they actively don't want people to use GH pages and are trying to discourage its use by making the UX more difficult and having purposefully wrong documentation to really throw you off.

**Reply by Ysgrathe** [reply]
That's crazy talk!
I mean, that would be like saying that Microsoft owning GitHub means they are now wanting to push people somehow towards their Azure based, paid for services for hosting, instead of GH pages.
And, you know... 
that's just... 
I mean...

<Ahem>
So is it only a matter of time until GH pages is completely removed, 'transferred' into an office 365 feature, or nerfed into unusability?
Is it actually worth setting up new sites using it, or should we all be starting to look for alternatives before the actual rug-pull happens?

**Comment by mbodm** [comment]
Hmm,

i don’t wanna say „_That happens when MS takes over…_“ since i don’t wanna sound like some a… ooops. 😁 

Seriously, i would have way less problems with such a design descission, if they also remove/change the docs. But this is just plain stupid confusing, imo.

**Comment by Mase3206** [comment]
Is there seriously no update on this from the devs?

Really?

I was genuinely thrilled to find out that GitHub Pages offers free hosting to static pages, because I was wanting to move away from website creators like Wordpress or Wix. All I encountered was problem after problem with no solution in sight, unhelpful documentation, an infuriating setup process involving me installing Linux on a *completely different device*, giving up, trying again, and just getting angry. I know I'm not the only one having issues, so ***why on earth isn't GitHub fixing them!?***

**Comment by mcoons** [comment]
I find this very frustrating.  First off, it was an easy click.  Second, why are you even forcing themes?  If I make a site then I probably already have it looking the way I want it to look.  

EASY PEAZY ANSWER WORKED FOR ME
I believe this was stated earlier, but...
from:    https://github.com/orgs/community/discussions/60961#

Go to your root folder of your repo
Add a file named config.yml
Add the following two lines in the file changing your site name:

name: MySite
theme: minima

Save and commit the changes.

Navigate to the Pages page
Select Branch as none - refresh
Select Branch as master - refresh
I was then able to Save

My page was running within a minute.

**Comment by Beelzenef** [comment]
still sad news

**Comment by robertbcalhoun** [comment]
@yoannchaudet , could you ask the documentation team to update https://pages.github.com/ , "Project Site" tab, to reflect the fact that the Theme Chooser has been removed? It's been well over a year since Github made this change, yet the marquee page for Github Pages still walks through setting up Github Pages via this unavailable feature. This results in a poor experience for new users of Github Pages. Thank you.

**Comment by jdies** [comment]
Ah interesting. I had good memories of github pages and was quite surprised when I tried using it right now.

**Comment by EngJPGodoi** [comment]
Is there any chance to it comeback to be usable?
And does anyone have an alternative option for GH Pages?

**Comment by mcoons** [comment]
I have published web based apps on itch.io and Firebase.  You can also obtain an AWS account and spin up an EC2 web server and self host apps that way also as long as it is minimal traffic.

GH pages do still work, they are just more of a pain to set up.  See my previous post on that.

**Comment by DalbarIII** [comment]
There is no access to the theme selector. It is no longer available. It is a shame that this course has inconsistencies like this. I know that we are learning and that we have to adjust to updates, but it is frustrating that the updates differ so much from the explanation. I wasted an entire day looking for the theme selector to get to the point that it no longer exists. Now we have to learn about github pages. another course.

**Reply by LiteBrite82** [reply]
@DalbarIII 
Which course are you taking?

**Comment by kairin** [comment]
okay so it is november 2024

![image](https://github.com/user-attachments/assets/1da18391-98be-4827-adeb-3887d538283d)

**Reply by kairin** [reply]
in case i keep coming back to this page because i forgot that i actually came to this page...

https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll

please remember to go to this docs instead...

**Reply by kairin** [reply]
else go to here:

https://docs.github.com/en/pages/quickstart

**Comment by LinuxMainframe** [comment]
Still not updated as of December 2024. Luckily, this page showed up pretty quickly due to traction gained by user input on said inconsistency. Adding my name in just to increase likelihood that new users will find this and use it to determine true course of action as opposed to the default Github help page.

TLDR, github needs to update its pages, especially surrounding setting up github pages.

**Comment by lizardcat** [comment]
Still not updated as of January 2025. Why were themes even removed in the first place? :(

**Comment by dcollien** [comment]
I'm here in March 2025 because I too wanted the theme chooser mentioned on https://pages.github.com/
Poor form, Github, poor form.

**Comment by The-Viper-One** [comment]
Just spent ages trying to figure what I was doing wrong as I could not see what I was clicking through marrying up with the documentation. Still not updated as of June 2025.
